onnxruntime/contrib_ops/rocm/math/gemm_float8.cu

```diff
@@ -0,0 +1,210 @@
+  auto b_shape = B->Shape();
+  ORT_ENFORCE(a_shape.NumDimensions() >= 2 && b_shape.NumDimensions() == 2);  // is in form of input @ weight
+  ORT_ENFORCE(a_shape[a_shape.NumDimensions() - 1] == b_shape[0]);            // k is compatiable
+
+  TensorShapeVector output_shape = a_shape.AsShapeVector();
+  output_shape[output_shape.size() - 1] = b_shape[b_shape.NumDimensions() - 1];
+  Tensor* Y = ctx->Output(0, output_shape);
+
+  ORT_ENFORCE(!transA_, "ROCm GemmFloat8 does not support input A transpose");
+  ORT_ENFORCE(dtype_ == onnx::TensorProto_DataType_FLOAT16, "ROCm GemmFloat8 only supports output float16");
+  ORT_ENFORCE(C == nullptr, "ROCm GemmFloat8 does not support bias input");
+  ORT_ENFORCE(scale_y == nullptr, "ROCm GemmFloat8 does not support output scaling");
+
+  if (A->IsDataType<Float8E4M3FN>()) {
+    return ComputeFp8Fp16Fp16<Float8E4M3FN>(ctx, A, scale_a, B, Y);
+  } else if (A->IsDataType<Float8E4M3FNUZ>()) {
+    return ComputeFp8Fp16Fp16<Float8E4M3FNUZ>(ctx, A, scale_a, B, Y);
+  } else if (B->IsDataType<Float8E4M3FN>()) {
+    return ComputeFp16Fp8Fp16<Float8E4M3FN>(ctx, A, B, scale_b, Y);
+  } else if (B->IsDataType<Float8E4M3FNUZ>()) {
+    return ComputeFp16Fp8Fp16<Float8E4M3FNUZ>(ctx, A, B, scale_b, Y);
+  }
+
+  return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Unhandled type combination of GemmFloat8");
+#endif
+}
+
+#if !defined(DISABLE_FLOAT8_TYPES)
+template <typename Fp8T>
+Status GemmFloat8::ComputeFp8Fp16Fp16(OpKernelContext* ctx, const Tensor* A, const Tensor* scale_a, const Tensor* B, Tensor* C) const {
+  ORT_ENFORCE(A->IsDataType<Fp8T>() && scale_a->IsDataType<float>() && B->IsDataType<MLFloat16>());
+
+  auto a_shape = A->Shape();
+  auto b_shape = B->Shape();
+
+  auto m = a_shape.Slice(0, a_shape.NumDimensions() - 1).Size();
+  auto k = a_shape[a_shape.NumDimensions() - 1];
+  auto n = b_shape[b_shape.NumDimensions() - 1];
+
+  onnxruntime::rocm::tunable::blas::GemmFloat8Params<Fp8T, MLFloat16, MLFloat16> params{};
+  params.tuning_ctx = GetTuningContext();
+  params.stream = ctx->GetComputeStream();
+  params.handle = GetRocblasHandle(ctx);
+  params.opa = transA_ ? tunable::blas::BlasOp::Trans : tunable::blas::BlasOp::NonTrans;
+  params.opb = transB_ ? tunable::blas::BlasOp::Trans : tunable::blas::BlasOp::NonTrans;
+
+  params.m = m;
+  params.n = n;
+  params.k = k;
+
+  params.a = static_cast<const Fp8T*>(A->DataRaw());
+  params.lda = transA_ ? m : k;
+  params.scale_a = alpha_;
+  params.scale_a_dev = static_cast<const float*>(scale_a->DataRaw());
+
+  params.b = static_cast<const MLFloat16*>(B->DataRaw());
+  params.ldb = transB_ ? k : n;
+  params.scale_b = 1.0f;         // NOTE: not used
+  params.scale_b_dev = nullptr;  // NOTE: not used
+
+  params.c = static_cast<MLFloat16*>(C->MutableDataRaw());
+  params.ldc = n;
+  params.scale_c = 1.0f;         // NOTE: not implemented
+  params.scale_c_dev = nullptr;  // NOTE: not implemented
+
+  // NOTE: transA is not implemented
+  if (transB_) {
```

Comment:
The comment shows transA is not implemented, but the code shows transB is not implemented, could you please comment more here?